Cython/Compiler/ModuleNode.py

```diff
@@ -1765,11 +1765,11 @@ def generate_dealloc_function(self, scope, code):
                     type_safety_check = ''
                 else:
                     type_safety_check = (
-                        ' & (!__Pyx_PyType_HasFeature(Py_TYPE(o), (Py_TPFLAGS_IS_ABSTRACT | Py_TPFLAGS_HEAPTYPE)))')
+                        ' & (int)(!__Pyx_PyType_HasFeature(Py_TYPE(o), (Py_TPFLAGS_IS_ABSTRACT | Py_TPFLAGS_HEAPTYPE)))')
 
                 type = scope.parent_type
                 code.putln(
-                    "if (CYTHON_COMPILING_IN_CPYTHON && ((%s < %d) & (Py_TYPE(o)->tp_basicsize == sizeof(%s))%s)) {" % (
+                    "if (CYTHON_COMPILING_IN_CPYTHON & (int)((%s < %d) && (Py_TYPE(o)->tp_basicsize == sizeof(%s))%s)) {" % (
```

Comment:
```suggestion
                    "if (CYTHON_COMPILING_IN_CPYTHON && ((int)(%s < %d) & (int)(Py_TYPE(o)->tp_basicsize == sizeof(%s))%s)) {" % (
```

I think this one was the wrong way round (i.e. `CYTHON_COMPILING_IN_CPYTHON` is known at compile time so order of evaluation really don't matter).

Other than that the cast to int seems like the cleanest way to silence the warning and preserve the existing logic